src/ol/style/IconImage.js

```diff
@@ -75,22 +75,18 @@ class IconImage extends EventTarget {
 
     /**
      * @private
-     * @type {boolean|undefined}
      */
     this.tainted_;
   }
 
   /**
    * @private
-   * @param {CanvasRenderingContext2D=} context A context with the image already drawn into.
    * @return {boolean} The image canvas is tainted.
    */
-  isTainted_(context) {
+  isTainted_() {
     if (this.tainted_ === undefined && this.imageState_ === ImageState.LOADED) {
-      if (!context) {
-        context = createCanvasContext2D(1, 1);
-        context.drawImage(this.image_, 0, 0);
-      }
+      const context = createCanvasContext2D(1, 1);
```

Comment:
Now that you do not pass an existing canvas any more, it would be better to have a dedicated canvas that you can reuse as long as it is not tainted. Especially in Safari, cleanup of unused canvases takes a while, so creating too many canvases should be avoided.

Reply:
+1, just done :-)